### ISSUE TITLE
Updating from 19.0.5 to 19.3 cooked my servertools

### ISSUE THREAD

**Issue by magejosh** [dmustanger/7dtd-ServerTools#263]
Now at the end of the bootup process when the server tools mod would confirm each modlet that was active, it instead only returns this single line. The commands do nothing now. I'm unsure how to fix my problem. I was hoping updating would fix the nonworking /sethome command since I really liked the lobby and jail ones. 

The server tools lines returned:

> 2020-12-20T21:46:33 142.778 INF [SERVERTOOLS] Checking for save directory C:\Pingperfect\Users\joshuam2\242468/ServerTools
2020-12-20T21:46:33 142.807 INF [SERVERTOOLS] Player data loaded
2020-12-20T21:46:33 142.923 INF ---------------------------------------------------------------
2020-12-20T21:46:33 142.924 INF [SERVERTOOLS] Verifying configuration file & Saving new entries
2020-12-20T21:46:33 142.924 INF ---------------------------------------------------------------
2020-12-20T21:46:33 142.933 INF [SERVERTOOLS] Error in API.GameStartDone: Could not resolve type with token 0100002a (from typeref, class/assembly HarmonyLib.Harmony, 0Harmony, Version=2.0.2.0, Culture=neutral, PublicKeyToken=null)
2020-12-20T21:46:33 142.933 INF StartGame done

**Comment by magejosh** [comment]
Worst part is that i couldn't even roll back to the previous version I was using, when i did nothing changed on the boot log, returns that same error again. I'm wondering if this isn't due to it being hosted on pingperfect.

**Comment by ObsComp** [comment]
Never seen this before...

I would recommend manually installing the files if you can. PingPerfect I believe has a one click install option that has been known to fail at times.

If I had to guess, your host blocked the 0Harmony.dll from being installed or updated possibly

**Comment by ObsComp** [comment]
Try and delete everything from ST and reinstall it fresh. You will lose anything saved such as home points but I imagine nothing is working anyways.

**Comment by magejosh** [comment]
I tried a delete of both the server tools folder and the mod folder, then reinstalled. That returned that same error code. With none of the modlet xml's being generated.
I opened a ticket with pingperfect to have them delete all and install on their end. They reported back they had done so.
Sadly i'm still seeing the same error in the console and none of the modlet xml's are being generated. It said it detected a previous version and backed up older settings. But if everything was deleted, idk how it's detecting a previous version.

**Comment by ObsComp** [comment]
It could not detect an older version unless the files were not actually deleted. you extract the Mods folder to your main installation directory. After starting the server it makes the ServerTools folder beside the Mods folder inside the same main installation directory. The xml files are all created there.

**Comment by ObsComp** [comment]
I have been told Ping tech guys are not thorough sometimes and miss that the dll files sometimes dont transfer over. If you access your ftp space and manually place the files, I have heard it will work. If the dll does not transfer in, you have to ask them to put that file in for you as that would be the culprit. If you can, send me a copy of your output log

**Comment by ObsComp** [comment]
2020-12-20T21:46:33 142.933 INF [SERVERTOOLS] Error in API.GameStartDone: Could not resolve type with token 0100002a (from typeref, class/assembly HarmonyLib.Harmony, 0Harmony, Version=2.0.2.0, Culture=neutral, PublicKeyToken=null)

That specific error if I am not mistaken is due to the 0Harmony.dll not getting updated or is missing. OR you have something else that is old running the old version of Harmony. Harmony was updated to version 2.0 + 

Are you running anything else that uses Harmony and needs updating?

**Comment by magejosh** [comment]
That is in fact what the conflict was. Wanted to come back here soon as I got it resolved. It was CPM was also running that same 0Harmony.dll and it's older version was creating the conflict it seems. Removed it and Server Tools worked again correctly. Added updated CPM and things are back the way they should be. Now to test /sethome. Thank You for your direction, I wouldn't have found that without your help.
